homeassistant/components/switchbot/cover.py

```diff
@@ -0,0 +1,181 @@
+"""Support for SwitchBot curtains."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.cover import (
+    ATTR_CURRENT_POSITION,
+    ATTR_POSITION,
+    DEVICE_CLASS_CURTAIN,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    SUPPORT_STOP,
+    CoverEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MAC, CONF_NAME, CONF_PASSWORD, CONF_SENSOR_TYPE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_CURTAIN,
+    CONF_RETRY_COUNT,
+    DATA_COORDINATOR,
+    DOMAIN,
+    MANUFACTURER,
+)
+from .coordinator import SwitchbotDataUpdateCoordinator
+
+# Initialize the logger
+_LOGGER = logging.getLogger(__name__)
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Switchbot curtain based on a config entry."""
+    coordinator: SwitchbotDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    if entry.data[CONF_SENSOR_TYPE] != ATTR_CURTAIN:
+        return
+
```

Comment:
It would be more efficient to filter `PLATFORMS` when calling `hass.config_entries.async_setup_platforms(entry, PLATFORMS)`

Reply:
Added if statements to remove platform from list before calling:

hass.config_entries.async_setup_platforms(entry, PLATFORMS)